### ISSUE TITLE
Can I learn the StreamPETR model using 3 images as inputs?

### ISSUE THREAD

**Issue by sjminInSwm** [exiawsh/StreamPETR#134]
Hello! @exiawsh 
I admire your good model!!! Thank you!!!!
I want to use 3 images in one device and succeeded in the inference and visualization by referring to the issue(#14 )

I want to learn StreamPETR by inputting **3 images using** a **nuscene data**
Can I train the StreamPETR model using 3 images as inputs?

Thank you for developing a good model!!!

**Comment by exiawsh** [comment]
@sjminInSwm Hi, thanks for your attention. Of course you can, but you should filter the ground truth.

**Comment by sjminInSwm** [comment]
Thank you for your response!! @exiawsh

Can I edit it in the next file to make pkl?

"tools/data_converter/nuscenes_converter.py"

**Comment by exiawsh** [comment]
Yes, you should modify the nuscenes converter , filter the camera views and ground-truth.

**Comment by sjminInSwm** [comment]
Thank you for your response! @exiawsh
I'll give it a try!!!

I have a another question..

I checked that the data called lidar2img is used to run Inference on StreamPETR model.

What this means is that we need lidar data when we inform StreamPETR model? But I know this model is camera-only-model.

The environment I have is an environment with 3 cameras without Lidar. (But I have Lidar Data to learn model.)

In conclusion,
1. Can I use StreamPETR model in environment without a Lidar?
2. if your answer is "yes" about "Question 1", Why use "lidar2img" to run a inference?

**Comment by exiawsh** [comment]
Our model don't use lidar data. But the boundingboxes for evaluation (in mmdet3d) are in the lidar coordinate. We use the lidar coordinate as the reference coordinate system for the ego car. Easy to unify coordinates in multiple camera coordinate systems.

**Comment by sjminInSwm** [comment]
Thank you for your quick response!! @exiawsh 

"https://github.com/woquyuna/AAA"
I am converting the onnx model by referring to the above github code.

In order to infer only three images, the input data were transformed with the following modifications.
![Screenshot from 2023-12-29 11-47-18](https://github.com/exiawsh/StreamPETR/assets/132979325/af4cbc85-b86e-4e52-91aa-de3121d9dce0)



The results of inference and visualization with the data sliced above are as follows.
![c5f58c19249d4137ae063b0e9ecd8b8e_pred](https://github.com/exiawsh/StreamPETR/assets/132979325/4aafbef8-05f1-4830-923e-d2cb97511a97)


I got the wrong detection result.

Can you suggest a direction to infer with 3 images?

**Comment by exiawsh** [comment]
You should finetune our model on 3 images setting.

**Comment by sjminInSwm** [comment]
> Yes, you should modify the nuscenes converter , filter the camera views and ground-truth.

Thanks! I'll try the way you taught me! Happy New Year!  :)  @exiawsh

**Comment by exiawsh** [comment]
@sjminInSwm Thank you! Happy New Year!

**Comment by sjminInSwm** [comment]
Happy New Year! @exiawsh 
I hope you will be full of happiness this year. :)

I'm trying to modify to learn using three images in the file  that makes the pkl file.

I modified this [part](https://github.com/exiawsh/StreamPETR/blob/16f2275c7335684223c0a51e130bd4a2bf15d8ad/tools/data_converter/nuscenes_converter.py#L208) to create a pkl file using only 3 cameras.

And I faced a problem trying to modify the ground truth.
I think the data coming through the lidar data is a box measured at a 360-degree angle!
So I think we should find a box in the area where 3 cameras are looking and make a ground truth! (The ground truth must be filtered because the box in the area that the camera cannot see provides a large loss value.) [I think this [part](https://github.com/exiawsh/StreamPETR/blob/16f2275c7335684223c0a51e130bd4a2bf15d8ad/tools/data_converter/nuscenes_converter.py#L174C21-L174C26) needs to be modified.]

But I can't think of the idea of filtering unused areas in the lidar data.

May I ask if you have an good idea to filter only boxes in areas that a particular camera sees?

**Comment by exiawsh** [comment]
@sjminInSwm You can project all the ground-truth to the 3 cameras. If the ground-truth does not hit on the 3 cameras, just filter it out. You can refer to our code to generate the 2D boxes labels and record the ground truth.

**Comment by sjminInSwm** [comment]
@exiawsh 

Your reply was very helpful to me!! Thank you!!!!!
